src/rio.c

```diff
@@ -108,19 +109,49 @@ void rioInitWithBuffer(rio *r, sds s) {
 
 /* Returns 1 or 0 for success/failure. */
 static size_t rioFileWrite(rio *r, const void *buf, size_t len) {
-    size_t retval;
+    if (!r->io.file.autosync) return fwrite(buf,len,1,r->io.file.fp);
 
-    retval = fwrite(buf,len,1,r->io.file.fp);
-    r->io.file.buffered += len;
+    size_t nwritten = 0;
+    while (len != nwritten) {
```

Comment:
let's add a comment describing that loop, what it does and why.
please see if you wanna add anything else on top of this suggestion:
```suggestion
    /* Incrementally write data to the file, avoid a single write larger than the autosync
       threshold (so that the kernel's buffer cache never has too many dirty pages at once). */
    while (len != nwritten) {
```